src/Deprecated/Engine/Engine/EventSource.cs

```diff
@@ -520,40 +520,57 @@ internal void RaiseAnyEvent(object sender, BuildEventArgs e)
         }
 
         /// <summary>
-        /// Raises the given event to all registered loggers. This method up-cast the events
+        /// Raises the given event to all registered loggers. This method down-casts the events
```

Comment:
Wait, isn't this an upcast? to the more specific type? That matches my understanding of the "direction".

Reply:
[StackOverflow](https://stackoverflow.com/questions/23414090/what-is-the-difference-between-up-casting-and-down-casting-with-respect-to-class) says upcasting is to a supertype. I wasn't familiar with either terminology; maybe just change it to "This method casts all the events extracted from the queue to their more specific form" or something like that?

Reply:
huh, it appears to be used that way in other dotnet repos too. Yet more fuel for my "don't use terminology that implies a direction" fire.

Reply:
I've replaced it with direction-free wording.

(Reminds me of my confusion every time someone talks about LEFT JOIN and RIGHT JOIN.)